### PR TITLE
Bullet Soft Body multithreaded implementation

### PR DESCRIPTION
**Pull request by Lopuska**
Hi @erwincoumans,
I wanted to share my parallel version of the soft body class specialization, It helps a lot to increase soft body instance count in a dynamic scene especially for games or other real-time applications.

Take your time to review it.
Enjoy and happy new year!

### CODE REVIEW

File: src/BulletSoftBody/btDefaultSoftBodySolverMt.cpp [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+		}
+	}
+};
+
+void btDefaultSoftBodySolverMt::updateSoftBodies()
+{
+	if (m_softBodySet.size() > 0)
+	{
+		SoftBodyUpdater loop(&m_softBodySet);
+		int grainSize = 2;
```
Comment (Lopuska) [inline src/BulletSoftBody/btDefaultSoftBodySolverMt.cpp]: Feel free to suggest any reasonable value for the task granularity. In the rigid body pipeline the value is hard-coded like here. 2 sounds reasonable.

**Comment by dyanikoglu** [comment]
Looking forward to see this on master branch, can confirm it works without problem.

**Comment by AGPX** [comment]
Hey, any chance to have also the 'btMultiBodyConstraintSolver' working in multithreading?

**Comment by Lopuska** [comment]
Hi @AGPX, I'll focus on that after this gets reviewed.
Also I need to do a deep study on it and see which path can be parallelized properly to get an effective performance boost.

**Comment by Lopuska** [comment]
Hi @erwincoumans 
Any luck on getting this reviewed eventually for future releases?
A nice advantage of this PR is also the possibility to use the already built-in multi-threaded rigid body pipeline automatically if you use a SoftBody world instance.

**Pull request closed without merging**